### PR TITLE
Fix layout & localization issues

### PR DESCRIPTION
**Pull request by diegoserranoa**
- Move Edit Score, Dismiss & Send buttons up on Feedback view to prevent being covered by keyboard.
- Fix score view layout on tablets
- Remove last CtA button on tablets if custom Thank You message not set
- Show buttons in last view if custom thankyou or social links are set
- Add CHANGELOG.md & LICENSE.md files to repo
- Update gitignore file to ignore *.iml files, which are IDE files
- Fix http response decoding on WootricRemoteRequestTask.java to show correct characters unicode characters
- Add log to requests and responses on WootricRemoteRequestTask.java
- Add Edit Score to LocalizedTexts
- Fix Edit Score button to show localized text
- Fix Submit text on tablets to show localized text
- Add getters for social share question and social share decline on Settings.java
- Fix final thankyou message. If custom message is set, show it. If not, show default thankyou message from localized texts
- Fix final thankyou message on tablets. If custom message is set, show it. If not, show default thankyou message from localized texts
- Use default hint color for feedback placeholder on wootric_survey_layout.xml
- Set buttons textSize from dimens files on wootric_survey_layout.xml
- Add dimens.xml to change font size for different screen densities
- Update settings test
- Remove .iml files

To test:
- First try without setting a WootricCustomThankYou. If you can try with different language better
- Then set a WootricCustomThankYou with just

```java
customThankYou.setText("Thank you");
customThankYou.setDetractorText("Detractor thank you");
customThankYou.setPassiveText("Passive thank you");
customThankYou.setPromoterText("Promoter thank you");
```
to verify that texts on the last view changes correctly

Then also set links to see the last CtA buttons

```java
customThankYou.setLinkText("Thank you link text");
customThankYou.setDetractorLinkText("Detractor link text");
customThankYou.setPassiveLinkText("Passive link text");
customThankYou.setPromoterLinkText("Promoter link text");
customThankYou.setLinkUri(Uri.parse("http://wootric.com/thank_you"));
customThankYou.setDetractorLinkUri(Uri.parse("http://wootric.com/detractor_thank_you"));
customThankYou.setPassiveLinkUri(Uri.parse("http://wootric.com/passive_thank_you"));
customThankYou.setPromoterLinkUri(Uri.parse("http://wootric.com/promoter_thank_you"));
customThankYou.setScoreInUrl(true);
customThankYou.setCommentInUrl(true);
```

### CODE REVIEW

**Comment by diegoserranoa** [comment]
Change 2 & 3 are straightforward. To test change 1:
For buttons to show up, one of the following should be set:
-WootricCustomThankYou
-FacebookPageId
-TwitterPage

To set WootricCustomThankYou:
WootricCustomThankYou customThankYou = new WootricCustomThankYou();
customThankYou.setText('Thank you");
customThankYou.setDetractorText("Detractor thank you");
customThankYou.setPassiveText("Passive thank you");
customThankYou.setPromoterText("Promoter thank you");
customThankYou.setLinkText("Thank you link text");
customThankYou.setDetractorLinkText("Detractor link text");
customThankYou.setPassiveLinkText("Passive link text");
customThankYou.setPromoterLinkText("Promoter link text");
customThankYou.setLinkUri(Uri.parse("http://wootric.com/thank_you"));
customThankYou.setDetractorLinkUri(Uri.parse("http://wootric.com/detractor_thank_you"));
customThankYou.setPassiveLinkUri(Uri.parse("http://wootric.com/passive_thank_you"));
customThankYou.setPromoterLinkUri(Uri.parse("http://wootric.com/promoter_thank_you"));
customThankYou.setScoreInUrl(true);
customThankYou.setCommentInUrl(true);
wootric.setCustomThankYou(customThankYou);

To set Facebook:
wootric.setFacebookPageId("123456");

To set Twitter:
wootric.setTwitterPage("wootric");

All of these should be set on MainActivity.java before wootric.survey();

**Comment by diegoserranoa** [comment]
To test all changes that require localization, just change the language on [https://app.wootric.com/user_settings/edit#!/survey-nps](url) and save. Survey should show the correct language and support unicode characters. 

Try different combinations:
-First without setting a WootricCustomThankYou. Test different languages & different devices, the more the better.
-Then set the WootricCustomThankYou with just 
`customThankYou.setText('Thank you");
customThankYou.setDetractorText("Detractor thank you");
customThankYou.setPassiveText("Passive thank you");
customThankYou.setPromoterText("Promoter thank you");`
to verify that text on the last view changes.
-Then also set
`customThankYou.setLinkText("Thank you link text");
customThankYou.setDetractorLinkText("Detractor link text");
customThankYou.setPassiveLinkText("Passive link text");
customThankYou.setPromoterLinkText("Promoter link text");
customThankYou.setLinkUri(Uri.parse("http://wootric.com/thank_you"));
customThankYou.setDetractorLinkUri(Uri.parse("http://wootric.com/detractor_thank_you"));
customThankYou.setPassiveLinkUri(Uri.parse("http://wootric.com/passive_thank_you"));
customThankYou.setPromoterLinkUri(Uri.parse("http://wootric.com/promoter_thank_you"));
customThankYou.setScoreInUrl(true);
customThankYou.setCommentInUrl(true);`
to other buttons on the last view.

**Comment by hachibu** [comment]
Delete all the iml files since you added it to the gitignore.

**Comment by hachibu** [comment]
Do these files need to be in the repo?
- build.gradle
- gradle/wrapper/gradle-wrapper.properties

File: CHANGELOG.md [hunk 1]
```diff
@@ -0,0 +1,9 @@
+## 2.4.1 (2016-05-04)
+
+### Fixed:
+
+- Move Edit Score, Dismiss & Send buttons up on Feedback view to prevent being covered by keyboard
+- Fix score view layout on tablets
+- Remove buttons in last view if custom Thank You message and Social links not set (tablets)
+- Fix custom thank you text to show correct text (tablet)
+- Add CHANGELOG.md & LICENSE.md
```
Comment (hachibu) [inline CHANGELOG.md]: Awesome, CHANGELOG is a nice addition! 👍

**Comment by diegoserranoa** [comment]
We do need those files!

**Comment by hachibu** [comment]
There is an issue with setting detractor and passive link uris. With the code you suggested to test the CtA:
- Detractor link goes to `http://wootric.com/thank_you`
- Passive link goes to `http://wootric.com/detractor_thank_you`

**Comment by hachibu** [comment]
I tested all the cases that you described and they work! 👍

**Pull request merged**